tools/codegen/main.go

```diff
@@ -37,3 +43,19 @@ func getOsArg() *string {
 	}
 	return &os.Args[1]
 }
+
+func writeToFile(fileName, content string) error {
+	// Open the file with write-only permission, create it if it doesn't exist and truncate its content if it exists
+	file, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.WriteString(content)
```

Comment:
knit: similar to this: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-278652_use_schemas/tools/check-changelog-entry-file/main.go#L35

there is os.WriteFile that might be a bit easier to use

Reply:
thanks, indeed does simplify the code, adjusted